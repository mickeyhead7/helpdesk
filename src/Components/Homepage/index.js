import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BodyHeader from '@Components/BodyHeader';
import ArticleList from '@Components/ArticleList';
import ConnectedTagList from '@Components/TagList';
import I18nProvider from '@packages/Components/i18n/Provider';
import Translate from '@packages/Components/i18n/Translate';
import enGB from './languages/en-GB';

const phrases = {
  'en-GB': enGB,
};

export const Homepage = ({ articles, locale, tags }) => (
  <I18nProvider locale={locale} phrases={phrases}>
    <BodyHeader>
      <h1>
        <Translate locale={locale} phrase="messages.welcome" />
      </h1>
    </BodyHeader>
    <section>
      <header>
        <h2>
          <Translate locale={locale} phrase="messages.latestArticles" />
        </h2>
      </header>
      <ArticleList articles={articles} />
    </section>
    <section>
      <ConnectedTagList tags={tags} />
    </section>
  </I18nProvider>
);

Homepage.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    node: PropTypes.shape({
      frontmatter: PropTypes.shape({
        excerpt: PropTypes.string.isRequired,
        path: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  })).isRequired,
  locale: PropTypes.string.isRequired,
  tags: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
};

const mapStateToProps = ({ i18n }) => ({
  locale: i18n.locale,
});

export default connect(mapStateToProps)(Homepage);
