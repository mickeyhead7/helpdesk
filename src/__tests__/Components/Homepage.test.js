import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Homepage } from '@Components/Homepage';

const article = {
  node: {
    frontmatter: {
      excerpt: 'This is the article excerpt.',
      date: '1970-01-01 00:00:00',
      title: 'Article title',
      path: '/article-path',
    },
    html: <div>Some html</div>,
  },
};

test('Homepage', () => {
  const props = {
    locale: 'en-GB',
    articles: [
      { ...article },
    ],
    tags: [
      'tag-one',
      'tag-two',
      'tag-three',
    ],
  };
  const component = shallow(<Homepage {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});
