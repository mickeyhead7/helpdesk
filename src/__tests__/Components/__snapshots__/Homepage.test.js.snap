// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage 1`] = `
<Connect(Provider)
  locale="en-GB"
  phrases={
    Object {
      "en-GB": Object {
        "messages": Object {
          "latestArticles": "Latest articles",
          "welcome": "Welcome to the helpdesk",
        },
      },
    }
  }
>
  <BodyHeader>
    <h1>
      <Connect(Translate)
        locale="en-GB"
        phrase="messages.welcome"
      />
    </h1>
  </BodyHeader>
  <section>
    <header>
      <h2>
        <Connect(Translate)
          locale="en-GB"
          phrase="messages.latestArticles"
        />
      </h2>
    </header>
    <ArticleList
      articles={
        Array [
          Object {
            "node": Object {
              "frontmatter": Object {
                "date": "1970-01-01 00:00:00",
                "excerpt": "This is the article excerpt.",
                "path": "/article-path",
                "title": "Article title",
              },
              "html": <div>
                Some html
              </div>,
            },
          },
        ]
      }
    />
  </section>
  <section>
    <Connect(TagList)
      tags={
        Array [
          "tag-one",
          "tag-two",
          "tag-three",
        ]
      }
    />
  </section>
</Connect(Provider)>
`;
